pkg/devfile/adapters/common/command.go

```diff
@@ -164,6 +170,28 @@ func validateCommand(data data.DevfileData, command common.DevfileCommand) (err
 	return
 }
 
+// validateCompositeCommand checks that the specified composite command is valid
+func validateCompositeCommand(data data.DevfileData, compositeCommand *common.Composite) error {
+	if compositeCommand.Group != nil && compositeCommand.Group.Kind == common.RunCommandGroupType {
+		return fmt.Errorf("composite commands of run Kind are not supported currently")
+	}
+
+	commandsMap := GetCommandsMap(data.GetCommands())
+
+	// Loop over the commands and validate that each command points to a command that's in the devfile
+	for _, command := range compositeCommand.Commands {
+		if command == compositeCommand.Id {
+			return fmt.Errorf("the composite command %q cannot reference itself", compositeCommand.Id)
```

Comment:
Good find. It ended up being a string comparison issue with non-lower case commands (like `buildAndEcho`). This is because the `compositeCommand.Id` value will be lower case, but the value of `command` in the subcommand array may not. So we have to call `strings.ToLower()`